packages/jellyfish-transaction/__tests__/script/dftx/dftx_icxorderbook/CreateOrder.test.ts

```diff
@@ -0,0 +1,144 @@
+import { SmartBuffer } from 'smart-buffer'
+import { OP_DEFI_TX } from '../../../../src/script/dftx'
+import { OP_CODES } from '../../../../src'
+import { toBuffer, toOPCodes } from '../../../../src/script/_buffer'
+import { CICXCreateOrder, ICXCreateOrder, ICXOrderType } from '../../../../src/script/dftx/dftx_icxorderbook'
+import BigNumber from 'bignumber.js'
+
+it('should bi-directional buffer-object-buffer', () => {
+  const fixtures = [
+    /**
+    * order : {
+    *   tokenFrom: '0',
+    *   chainTo: 'BTC',
+    *   ownerAddress: '2N82JqppeCXR97CbrV18ZyJy4gD6swrcetY',
+    *   receivePubkey: '037f9563f30c609b19fd435a19b8bde7d6db703012ba1aba72e9f42a87366d1941',
+    *   amountFrom: new BigNumber(15),
+    *   orderPrice: new BigNumber(0.01)
+    * }
+    */
+    '6a4c5d4466547831010017a914a21888038057490b84aab333846aa568839520d68721037f9563f30c609b19fd435a19b8bde7d6db703012ba1aba72e9f42a87366d1941002f685900000000002f68590000000040420f0000000000400b0000',
+    /**
+     * order : {
+     *  chainFrom: 'BTC',
+     *  tokenTo: '0',
+     *  ownerAddress: '2N7PZSFc5pbqwuXoShQKHzRBARwE7bNkvq8',
+     *  amountFrom: new BigNumber(2),
+     *  orderPrice: new BigNumber(1000)
+     * }
+     */
+    '6a3c4466547831020017a9149b2552a44b95afa2d435e416f8c16cfd12e97efd870000c2eb0b0000000000c2eb0b0000000000e8764817000000400b0000',
+    /**
+     * order :  {
+     *  tokenFrom: '0',
+     *  chainTo: 'BTC',
+     *  ownerAddress: '2N7PZSFc5pbqwuXoShQKHzRBARwE7bNkvq8',
+     *  receivePubkey: '037f9563f30c609b19fd435a19b8bde7d6db703012ba1aba72e9f42a87366d1941',
+     *  amountFrom: new BigNumber(15),
+     *  orderPrice: new BigNumber(0.01)
+     * }
+     */
+    '6a4c5d4466547831010017a9149b2552a44b95afa2d435e416f8c16cfd12e97efd8721037f9563f30c609b19fd435a19b8bde7d6db703012ba1aba72e9f42a87366d1941002f685900000000002f68590000000040420f0000000000400b0000'
+
+  ]
+
+  fixtures.forEach(hex => {
+    const stack = toOPCodes(
+      SmartBuffer.fromBuffer(Buffer.from(hex, 'hex'))
+    )
+    const buffer = toBuffer(stack)
+    expect(buffer.toString('hex')).toStrictEqual(hex)
+    expect((stack[1] as OP_DEFI_TX).tx.type).toStrictEqual(0x31)
+  })
+})
+
+describe('CreateOrder with receivePubkey', () => {
+  const header = '6a4c5d4466547831' // OP_RETURN PUSH_DATA(44665478, 31)
+  const data = '010017a914a21888038057490b84aab333846aa568839520d68721037f9563f30c609b19fd435a19b8bde7d6db703012ba1aba72e9f42a87366d1941002f685900000000002f68590000000040420f0000000000400b0000'
+  const createOrder: ICXCreateOrder = {
+    orderType: ICXOrderType.INTERNAL,
```

Comment:
hmm. can use `ICXOrderType` from icxorderbook ?
like
https://github.com/DeFiCh/jellyfish/blob/ba47d8a948789a5ac546895d8f8a4c64a22d8780/packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_submit_dfc_htlc.test.ts#L107

Reply:
In that case, the enum value needs to be a number
```
export enum ICXOrderType {
  /** type for DFI/BTC orders */
  INTERNAL = 1,
  /** type for BTC/DFI orders */
  EXTERNAL = 2
}
```

Reply:
my bad.